packages/backend-core/src/cache/docWritethrough.ts

```diff
@@ -0,0 +1,140 @@
+import BaseCache from "./base"
+import { getDocWritethroughClient } from "../redis/init"
+import { AnyDocument, Database, LockName, LockType } from "@budibase/types"
+import * as locks from "../redis/redlockImpl"
+
+import { JobQueue, createQueue } from "../queue"
+import * as context from "../context"
+import * as dbUtils from "../db"
+
+let CACHE: BaseCache | null = null
+async function getCache() {
+  if (!CACHE) {
+    const client = await getDocWritethroughClient()
+    CACHE = new BaseCache(client)
+  }
+  return CACHE
+}
+
+interface ProcessDocMessage {
+  tenantId: string
+  dbName: string
+  docId: string
+  cacheKeyPrefix: string
+}
+
+export const docWritethroughProcessorQueue = createQueue<ProcessDocMessage>(
+  JobQueue.DOC_WRITETHROUGH_QUEUE
+)
+
+docWritethroughProcessorQueue.process(async message => {
+  const { tenantId, cacheKeyPrefix } = message.data
+  await context.doInTenant(tenantId, async () => {
+    const lockResponse = await locks.doWithLock(
+      {
+        type: LockType.TRY_ONCE,
+        name: LockName.PERSIST_WRITETHROUGH,
+        resource: cacheKeyPrefix,
+        ttl: 15000,
+      },
+      async () => {
+        await persistToDb(message.data)
+        console.log("DocWritethrough persisted", { data: message.data })
+      }
+    )
+
+    if (!lockResponse.executed) {
+      if (
+        lockResponse.reason !==
+        locks.UnsuccessfulRedlockExecutionReason.LockTakenWithTryOnce
+      ) {
+        console.error("Error persisting docWritethrough", {
+          data: message.data,
+        })
+        throw "Error persisting docWritethrough"
+      }
+      console.log(`Ignoring redlock conflict in write-through cache`)
```

Comment:
Could you explain what's going on here? I don't know what `locks.UnsuccessfulRedlockExecutionReason.LockTakenWithTryOnce` means in practice, and it seems scary to ignore any errors in this code.

Reply:
Removed, after our last discussion